grpc/src/main/java/com/linecorp/armeria/server/grpc/DefaultUnframedGrpcErrorHandler.java

```diff
@@ -0,0 +1,276 @@
+                                                                   .addInt(GrpcHeaderNames.GRPC_STATUS,
+                                                                           grpcCode.value())
+                                                                   .build();
+            final HttpData content;
+            try (TemporaryThreadLocals ttl = TemporaryThreadLocals.acquire()) {
+                final StringBuilder msg = ttl.stringBuilder();
+                msg.append("grpc-code: ").append(grpcCode.name());
+                if (grpcMessage != null) {
+                    msg.append(", ").append(grpcMessage);
+                }
+                if (cause != null && ctx.config().verboseResponses()) {
+                    msg.append("\nstack-trace:\n").append(Exceptions.traceText(cause));
+                }
+                content = HttpData.ofUtf8(msg);
+            }
+            return HttpResponse.of(responseHeaders, content);
+        };
+    }
+
+    /**
+     * Returns a rich error JSON response based on Google APIs.
+     * Please refer <a href="https://cloud.google.com/apis/design/errors#error_model">Google error model</a>
+     * @param statusMappingFunction The function which maps the {@link Throwable} or gRPC {@link Status} code
+     *                              to an {@link HttpStatus} code.
+     */
+    static UnframedGrpcErrorHandler ofRichJson(UnframedGrpcStatusMappingFunction statusMappingFunction) {
+        final UnframedGrpcStatusMappingFunction mappingFunction =
+                requireNonNull(statusMappingFunction, "statusMappingFunction")
+                        .orElse(UnframedGrpcStatusMappingFunction.of());
+        return (ctx, status, response) -> {
+            final Code grpcCode = status.getCode();
+            @Nullable
+            final String grpcMessage = status.getDescription();
+            @Nullable
+            final Throwable cause = getThrowableFromContext(ctx);
+            final HttpStatus httpStatus = mappingFunction.apply(ctx, status, cause);
+            final HttpHeaders trailers = !response.trailers().isEmpty() ?
+                                         response.trailers() : response.headers();
+            @Nullable
+            final String grpcStatusDetailsBin = trailers.get(GrpcHeaderNames.GRPC_STATUS_DETAILS_BIN);
+            final ResponseHeaders responseHeaders = ResponseHeaders.builder(httpStatus)
+                                                                   .contentType(MediaType.JSON_UTF_8)
+                                                                   .addInt(GrpcHeaderNames.GRPC_STATUS,
+                                                                           grpcCode.value())
+                                                                   .build();
+            final ImmutableMap.Builder<String, Object> messageBuilder = ImmutableMap.builder();
+            messageBuilder.put("code", httpStatus.code());
```

Comment:
It seems like the code is `google.rpc.Code`. 
```
  // A simple error code that can be easily handled by the client. The
  // actual error code is defined by `google.rpc.Code`.
  int32 code = 1;
```

Reply:
Yeah, that code is `google.rpc.Status.code`.
And the actual error code is `google.rpc.Code` according to the document.
However, we are assigning `HttpStatus` code. Did I miss something? 🤔

Reply:
So the real problem here is that google seems to use a different schema for JSON HTTP APIs. So I think maybe we should not refer to the original error proto message. Instead we should use 
// This message defines the error schema for Google's JSON HTTP APIs.
```
message Error {
  // Deprecated. This message is only used by error format v1.
  message ErrorProto {}
  // This message has the same semantics as `google.rpc.Status`. It uses HTTP
  // status code instead of gRPC status code. It has extra fields `status` and
  // `errors` for backward compatibility with [Google API Client
  // Libraries](https://developers.google.com/api-client-library).
  message Status {
    // The HTTP status code that corresponds to `google.rpc.Status.code`.
    int32 code = 1;
    // This corresponds to `google.rpc.Status.message`.
    string message = 2;
    // Deprecated. This field is only used by error format v1.
    repeated ErrorProto errors = 3;
    // This is the enum version for `google.rpc.Status.code`.
    google.rpc.Code status = 4;
    // This corresponds to `google.rpc.Status.details`.
    repeated google.protobuf.Any details = 5;
  }
  // The actual error payload. The nested message structure is for backward
  // compatibility with [Google API Client
  // Libraries](https://developers.google.com/api-client-library). It also
  // makes the error more readable to developers.
  Status error = 1;
}
```
according to the [document](https://cloud.google.com/apis/design/errors#error_model)?

Reply:
Oops, I was looking at a different section. 😅 Yeah, indeed it's an HTTP status code in [HTTP Mapping](https://cloud.google.com/apis/design/errors#http_mapping). Thanks!